utils/src/lib.rs

```diff
@@ -55,6 +56,32 @@ pub fn union(a: &[u8], b: &[u8]) -> Vec<u8> {
     union
 }
 
+/// Concatenate a namespace and a message, prepended by a varint encoding of the namespace length.
+///
+/// This produces a unique byte sequence (i.e. no collisions) for each `(namespace, msg)` pair.
+pub fn union_unique(namespace: &[u8], msg: &[u8]) -> Vec<u8> {
+    // Get the length of the namespace as varint.
+    let mut len = namespace.len() as u64;
+
+    // Even if the namespace is empty, we still need to encode the length with a single zero byte.
```

Comment:
If we are implementing our own version of varint here, I'd suggest we either:
* Expose it outside of this function for others to use
* Replace it with some external implementation of varint.